src/app/donation-start/donation-start-form/donation-start-form.component.ts

```diff
@@ -993,6 +993,27 @@ export class DonationStartFormComponent implements OnDestroy, OnInit, AfterViewI
       return;
     }
 
+    // Check if matching has expired without the timer firing (e.g. iOS issue or page backgrounded)
+    if (this.donation && this.donation.matchReservedAmount > 0 && this.donation.createdTime) {
+      const expiryTime = environment.reservationMinutes * 60000 + new Date(this.donation.createdTime).getTime();
+      const now = Date.now();
```

Comment:
I think this is good, not quite sure if it's better to use the local clock like this rather than attempting to confirm the donation and then detecting the match expired error coming back from matchbot and prompting to remove the matching then. Would allow the donor another couple of minutes of matching and removing reliance on the accuracy of the clock in the donor's device. Probably doesn't make much difference either way.